qbraid_qir/squin/visitor.py

```diff
@@ -0,0 +1,562 @@
+# Copyright 2025 qBraid
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# pylint: disable-all
+# type: ignore
+
+import os
+import re
+import struct
+from dataclasses import dataclass, field
+from plistlib import InvalidFileException
+from typing import Any
+
+import pyqir
+from bloqade import qubit
+from bloqade.squin import gate, kernel, qalloc
+from kirin import ir, lowering, types
+from kirin.dialects import func, ilist, py
+from kirin.rewrite import CFGCompactify, Walk
+
+
+def load(
+    module: str | pyqir.Module,
+    kernel_name: str = "main",
+    dialects: ir.DialectGroup = kernel,
```

Comment:
Can we use `kwargs` here instead and list them out instead of a large number of parameters? Unless `load` is an `abstractmethod` which requires these args in its definition?